mozilla-release/browser/base/content/browser.js

```diff
@@ -7396,6 +7399,44 @@ var gRemoteTabsUI = {
   }
 };
 
+// Small utility handler which propagates current tab "privateness" property as
+// available to CSS selectors through #navigator-toolbox[on_private_tab]
+const _gPrivateTabUI = {
+  init: function PrivateTabUI_init() {
+    this._updateNavToolbox();
+    for (let eventName in this._tabContainerEvents) {
+      gBrowser.tabContainer.addEventListener(eventName, this);
```

Comment:
It works correctly. Object passed to `addEventListener ` is called through `handleEvent ` function.
Inside `handleEvent ` we pass control to appropriate handler calling it with context of `_gPrivateTabUI`, because `_tabContainerEvents` here is more a map container than an individual object.